airflow/providers/google/provider.yaml

```diff
@@ -95,7 +95,7 @@ dependencies:
   - google-cloud-dataflow-client>=0.8.2
   - google-cloud-dataform>=0.5.0
   - google-cloud-dataplex>=1.4.2
-  - google-cloud-dataproc>=5.4.0
+  - google-cloud-dataproc>=5.5.0
```

Comment:
Is there a reason why 5.4.x is not compatible with the changes added in this PR?

Reply:
The new fields (min_num_instances and instance_flexibility_policy) are available from 5.5.0

Reply:
Would it be possible to detect the version and only allow them in 5.5.0, and keep compatibility to the lower version? (Is it worthwhile?)

Reply:
It is possible to detect the version using `__version__` but don't think it would be a good idea to do so.

Reply:
Can you elaborate?

Reply:
> Would it be possible to detect the version and only allow them in 5.5.0, and keep compatibility to the lower version? (Is it worthwhile?)

I think not worthwile -> generally the "cloud" services work best with newest versions of libraries, and from the point of view of Airlfow "google provider" is the only consumer of this library. 

Generally speaking - when we are talking about providers, as long as we already use higher version of the library in main constraints (we do https://github.com/apache/airflow/blob/constraints-main/constraints-3.10.txt#L321C1-L322C1)  - bumping minimum version of pretty much any dependency (except some really popular and possibly not easy to upgrade to) is safe. Users can always use earlier version of provider if they require the dependency in lower version, and actually in most installation cases this will happen automatically - provider will be downgraded in case user will want to install non-constraint compatible version of the dependency.

So pretty much in any case where we have a reason to bump min version of dependeny in a provider, we should do it (unless we have a very good reason not to - but this should be exceptional case).

Reply:
Agree with above. Also, can someone please trigger the remaining workflows.